delta/app/src/test/scala/ch/epfl/bluebrain/nexus/delta/MainSuite.scala

```diff
@@ -68,62 +69,44 @@ object MainSuite {
-      System.clearProperty("plugins.elasticsearch.credentials.username")
-      System.clearProperty("plugins.elasticsearch.credentials.password")
-
-      System.clearProperty("plugins.graph-analytics.enabled")
-      System.clearProperty("plugins.search.enabled")
-
-      System.clearProperty("plugins.search.indexing.mapping")
-      System.clearProperty("plugins.search.indexing.query")
-      System.clearProperty("plugins.search.indexing.resource-types")
-      ConfigImpl.reloadSystemPropertiesConfig()
-    }
+    private def initConfig(postgres: PostgresContainer, elastic: ElasticSearchContainer): IO[Map[String, String]] =
+      IO.blocking {
+        val resourceTypesFile = Files.createTempFile("resource-types", ".json")
+        Files.writeString(resourceTypesFile, """["https://neuroshapes.org/Entity"]""")
+        val mappingFile       = Files.createTempFile("mapping", ".json")
+        Files.writeString(mappingFile, "{}")
+        val queryFile         = Files.createTempFile("query", ".json")
+        Files.writeString(
+          queryFile,
+          """CONSTRUCT { {resource_id} <http://schema.org/name> ?name } WHERE { {resource_id} <http://localhost/name> ?name }"""
+        )
+
+        Map(
+          "app.defaults.database.access.host"                  -> postgres.getHost,
+          "app.defaults.database.access.port"                  -> postgres.getMappedPort(5432).toString,
+          "app.database.tables-autocreate"                     -> "true",
+          "app.defaults.database.access.username"              -> PostgresUser,
+          "app.default.database.access.password"               -> PostgresPassword,
+          "akka.actor.testkit.typed.throw-on-shutdown-timeout" -> "false",
+          "plugins.elasticsearch.base"                         -> s"http://${elastic.getHost}:${elastic.getMappedPort(9200)}",
+          "plugins.elasticsearch.credentials.username"         -> "elastic",
+          "plugins.elasticsearch.credentials.password"         -> "password",
+          //TODO Investigate how to remove this property from the config
+          "plugins.elasticsearch.disable-metrics-projection"   -> "true",
+          "plugins.graph-analytics.enabled"                    -> "true",
+          "plugins.search.enabled"                             -> "true",
+          "plugins.search.indexing.resource-types"             -> resourceTypesFile.toString,
+          "plugins.search.indexing.mapping"                    -> mappingFile.toString,
+          "plugins.search.indexing.query"                      -> queryFile.toString
+        )
+      }
 
     // Start the necessary containers
     private def resource() =
       for {
         postgres <- PostgresContainer.resource(PostgresUser, PostgresPassword)
         elastic  <- ElasticSearchContainer.resource()
-        _        <- Resource.make(acquire(postgres, elastic))(_ => release)
+        _        <- SystemPropertyOverride(initConfig(postgres, elastic))
```

Comment:
Introducing a a Cats Resource to handle the setup and teardown of system properties